app/Filament/Resources/ApiTokenResource.php

```diff
@@ -0,0 +1,158 @@
+    {
+        return [
+            Card::make()->schema([
+                TextInput::make('name')
+                    ->label('Name')
+                    ->unique(ignoreRecord: true)
+                    ->maxLength(100)
+                    ->required()
+                    ->autocomplete(false),
+
+                CheckboxList::make('abilities')
+                    ->label('Abilities')
+                    ->options([
+                        'results:read' => 'Read results',
+                        'speedtests:run' => 'Run speedtest',
+                        'ookla:list-servers' => 'List servers',
+                    ])
+                    ->required()
+                    ->bulkToggleable()
+                    ->descriptions([
+                        'results:read' => 'Allow this token to read results.',
+                        'speedtests:run' => 'Allow this token to run speedtests.',
+                        'ookla:list-servers' => 'Allow this token to list servers.',
+                    ]),
+
+                DateTimePicker::make('expires_at')
+                    ->label('Expires at')
+                    ->nullable()
+                    ->native(false)
+                    ->helperText('Leave empty for no expiration'),
+            ]),
+        ];
+    }
+
+    public static function form(Form $form): Form
+    {
+        return $form->schema(static::getTokenFormSchema());
+    }
+
+    public static function table(Table $table): Table
+    {
+        return $table
+            ->query(PersonalAccessToken::query()->where('tokenable_id', Auth::id()))
+            ->columns([
+                TextColumn::make('name')->searchable(),
+                TextColumn::make('abilities')->badge(),
+                IconColumn::make('is_valid')
+                    ->label('Valid')
+                    ->boolean()
+                    ->toggleable()
+                    ->state(fn ($record) => $record->expires_at === null || $record->expires_at->isFuture()),
+                TextColumn::make('created_at')
+                    ->dateTime(config('app.datetime_format'))
+                    ->timezone(config('app.display_timezone'))
+                    ->toggleable()
+                    ->sortable()
+                    ->alignEnd(),
+                TextColumn::make('last_used_at')
+                    ->dateTime(config('app.datetime_format'))
+                    ->timezone(config('app.display_timezone'))
+                    ->toggleable()
+                    ->toggledHiddenByDefault()
+                    ->sortable()
+                    ->alignEnd(),
+                TextColumn::make('expires_at')
+                    ->dateTime(config('app.datetime_format'))
+                    ->timezone(config('app.display_timezone'))
+                    ->toggleable()
+                    ->sortable()
+                    ->alignEnd(),
+            ])
+            ->filters([
+                TernaryFilter::make('valid')
```

Comment:
Instead of "valid" let's use the same language the framework uses and call it "expired". We can do filters for...

- "All tokens"
- "Active tokens" (active is a good antonym for expired)
- "Expired tokens"